plugins/copr.py

```diff
@@ -64,6 +64,20 @@ def linux_distribution():
                         # variable assignment
                         pass
                 return (os_release_data['NAME'], os_release_data['VERSION_ID'], None)
+        # Fallback for like() too
+
+        def like():
+            with open('/etc/os-release') as os_release_file:
+                os_release_data = {}
+                for line in os_release_file:
+                    try:
+                        os_release_key, os_release_value = line.rstrip().split('=')
+                        os_release_data[os_release_key] = os_release_value.strip('"')
+                    except ValueError:
+                        # Skip empty lines and everything that is not a simple
+                        # variable assignment
+                        pass
+                return os_release_data['ID_LIKE']
```

Comment:
This is obviously wrong, the `except` execution branch handles the case of failing to import `linux_distribution`, which it then defines. If you wanna do this, best probably to import the `like` separately and then handle a failed import. When was the `like` added and which distros still don't have it?

Also the code above is somewhat complicated and possibly handles cases which are long since not relevant, so it could probably use some simplification (instead of complicating it more by adding to it).

Reply:
AlmaLinux seems to not have `python3-distro` installed out of the box, so I tried to add a case for it in case of a failed import.

So should I simply ignore this import or add a seperate import for this?

Reply:
> AlmaLinux seems to not have `python3-distro` installed out of the box, so I tried to add a case for it in case of a failed import.

It is in Requires https://github.com/rpm-software-management/dnf-plugins-core/blob/master/dnf-plugins-core.spec#L149 so installation of dnf-plugins-core should install it. If it is not there fix the requirements in Alma rather than doing workarounds here in a code.